pandas/core/arrays/boolean.py

```diff
@@ -126,7 +126,12 @@ def __from_arrow__(
             bool_arr = BooleanArray._from_sequence(np.array(arr))
             results.append(bool_arr)
 
-        return BooleanArray._concat_same_type(results)
+        if not results:
```

Comment:
```suggestion
        if len(results) == 0:
```

? (I find that more explicit / readable)

Reply:
I can simply revert https://github.com/pandas-dev/pandas/pull/41052/commits/f4a2c8c90b5e8c0218869c5eff30354603b3bca5 (did do that commit as @phofl suggested)

Reply:
Whoops, missed that comment. Sorry for the contradicting feedback ;)

(let's leave it for now then, something to discuss / agree on as pandas team)

Reply:
Saw this on a few prs in the past that either not was used or it was recommended to use not, hence my feedback, sorry if I misinterpreted this